Feat - Price formatter classes

alecritson opened the pull request:
This PR intends to move the price formatting into a separate class which you can swap out in the configuration.

Currently formatting of prices are tied to the Lunar implementation and we should be allowing developers to provide their own methods.

---

Currently I'm not too sure on the implementation 🤔 as it stands it's just a copy of the existing methods and moved to a class reference, but I'm not sure how correct...

glennjacobs commented:
I think the function parameters probably need to be flexible, as if someone isn't using PHP's NumberFormatter class it wouldn't make sense.

alecritson commented:
Updated to decouple the arguments from the price datatype so it's up to the formatter itself what it supports. Also added a `prices` reference which should hopefully provide enough guidance 🤞

glennjacobs commented on docs/core/reference/pricing.md:
````diff
@@ -0,0 +1,210 @@
+];
+```
+
+When you retrieve a `Lunar\Models\Price` model, you will have access to the `->price` attribute which will return
````
Is it worth noting here that this isn't the only place a `price` type will be encountered, e.g. order lines, carts, etc.

glennjacobs commented:
Do we need to add something to the upgrade guide for the config value?

glennjacobs commented on docs/core/reference/pricing.md:
````diff
@@ -0,0 +1,210 @@
+);
+```
+
+### Creating a custom formatter
````
Can we use H2 headers where suitable so the TOC on the right shows the sections of the page?

glennjacobs reviewed (changes requested):
Left some comments for consideration

alecritson commented:
> Do we need to add something to the upgrade guide for the config value?

It's not a requirement to add it when updating, Lunar would still function. It would only be relevant if the developer was to change the class and they'd get that info from the other docs.

glennjacobs commented:
> > Do we need to add something to the upgrade guide for the config value?
> 
> It's not a requirement to add it when updating, Lunar would still function. It would only be relevant if the developer was to change the class and they'd get that info from the other docs.

No, but it is a change. Are we only showing breaking changes in the upgrade guide?

alecritson commented:
> > > Do we need to add something to the upgrade guide for the config value?
> > 
> > 
> > It's not a requirement to add it when updating, Lunar would still function. It would only be relevant if the developer was to change the class and they'd get that info from the other docs.
> 
> No, but it is a change. Are we only showing breaking changes in the upgrade guide?

My take is that the upgrade guide is for steps which are required for the upgrade to be successful. For example Laravel has this statement:

> We also encourage you to view the changes in the laravel/laravel [GitHub repository](https://github.com/laravel/laravel). While many of these changes are not required, you may wish to keep these files in sync with your application. Some of these changes will be covered in this upgrade guide, but others, such as changes to configuration files or comments, will not be.
